File: Java/PGS_두개뽑아서더하기.java
import java.util.*;
import java.io.*;
class Solution {
    public int[] solution(int[] numbers) {
        Set<Integer> hashset = new HashSet<>();
        
        for(int i = 0 ; i <numbers.length ; i++){
            for(int j = i+1 ; j < numbers.length ; j++){
                hashset.add(numbers[i]+numbers[j]);       
            }
        }
        Iterator iterator = hashset.iterator();
        int answer[] = new int[hashset.size()];
        
        for(int i = 0 ; i < hashset.size(); i++){
            answer[i] = (int)iterator.next();
        }
        Arrays.sort(answer);
        return answer;
    }
}

File: JS/PGS_두개뽑아서더하기.js
function solution(numbers) {
  var answer = [];

  for (let i = 0; i < numbers.length; i++) {
    for (let j = i + 1; j < numbers.length; j++) {
      let sum = numbers[i] + numbers[j];
      if (answer.indexOf(sum) === "") {
        answer.push(sum);
      }
    }
  }
  answer.sort((a, b) => a - b);
  return answer;
}

// if(answer.find(item => item === sum ) != undefined) continue; 도 가능!
